tools/clang/lib/SPIRV/RawBufferMethods.cpp

```diff
@@ -284,44 +285,46 @@ SpirvInstruction *RawBufferHandler::processTemplatedLoadFromBuffer(
   // aligned like their field with the largest alignment.
   // As a result, there might exist some padding after some struct members.
   if (const auto *structType = targetType->getAs<RecordType>()) {
-    const auto *decl = structType->getDecl();
+    LowerTypeVisitor lowerTypeVisitor(astContext, theEmitter.getSpirvContext(),
+                                      theEmitter.getSpirvOptions(), spvBuilder);
+    const StructType *spvType = dyn_cast<StructType>(lowerTypeVisitor.lowerType(
+        targetType, theEmitter.getSpirvOptions().sBufferLayoutRule, llvm::None,
+        SourceLocation()));
```

Comment:
This should probably be the Load call's source location, right?

Reply:
I'm not sure. I don't expect this to be the final source location when the spir-v is emitted because the result is used for iterating, but is not used or kept. I would expect the source location to be irrelevant to the final spir-v.

However, do not know that for sure, so I updated it. The source location of the load is not correct either, because the type is not defined at the position of the load. I used the source location of the tag decl associated with the struct. I believe one should always exist.